File: app/src/main/java/io/ipoli/android/note/data/Note.java
package io.ipoli.android.note.data;

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 9/21/16.
 */
public class Note {

    public enum NoteType {TEXT, INTENT, URL}

    private String noteType;
    private String text;
    private String data;

    public Note() {

    }

    public Note(NoteType noteType, String text, String data) {
        this.noteType = noteType.name();
        this.text = text;
        this.data = data;
    }

    public Note(String text) {
        this.noteType = NoteType.TEXT.name();
        this.text = text;
        this.data = "";
    }

    public String getNoteType() {
        return noteType;
    }

    public NoteType getNoteTypeValue() {
        return NoteType.valueOf(noteType);
    }

    public String getText() {
        return text;
    }

    public String getData() {
        return data;
    }

    public void setNoteType(String noteType) {
        this.noteType = noteType;
    }

    public void setText(String text) {
        this.text = text;
    }

    public void setData(String data) {
        this.data = data;
    }
}

File: app/src/main/java/io/ipoli/android/quest/data/Note.kt
package io.ipoli.android.quest.data

import io.realm.RealmObject

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 8/19/17.
 */
open class Note : RealmObject {
    enum class NoteType {

        TEXT, INTENT, URL
    }

    var noteType: String? = null
    var text: String? = null
    var data: String? = null

    constructor()

    constructor(noteType: NoteType, text: String, data: String) {
        this.noteType = noteType.name
        this.text = text
        this.data = data
    }

    constructor(text: String) {
        this.noteType = NoteType.TEXT.name
        this.text = text
        this.data = ""
    }

    val noteTypeValue: NoteType
        get() = NoteType.valueOf(noteType!!)
}
